### PR TITLE
Add docs regarding creating and checking tripal jobs.

### LINKED ISSUE #83: Document how to create jobs and use callables.
Exactly as it says 🤷‍♀️ We need to document how module developers can submit tripal jobs.

See branch tv4gAPI-issue83-createTripalJobs

### PR DESCRIPTION
**Pull request by laceysanderson**
Issue #83 

With PR https://github.com/tripal/tripal/pull/1917 we added expanded support for callables in the tripal job callback. This PR documents that and also basic programmatic interactions with tripal jobs since apparently those did not already exist 🙈 

Here are the rendered docs:
image

### CODE REVIEW

File: docs/dev_guide/module_dev/jobs.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+
+Jobs (longer-running tasks)
+=============================
+
+Often times whne dealing with biological data, the dataset size means that actions need to be done outside the page load. This is supported by Tripal through a jobs system that allows you to
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
Often times when dealing with biological data, the dataset size means that actions need to be done outside the page load. This is supported by Tripal through a jobs system that allows you to
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+1. Submit jobs on behalf of a user during a page load process.
+2. The job is registered with the Tripal Jobs system.
+3. Either you manually run it on the command line via drush or you set up a daemon or cron job to do so automatically. This allows the job longer execution time and to take advantage more of server resources.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
3. Either you manually run it on the command line via drush or you set up a daemon or cron job to do so automatically. This allows the job longer execution time and to take better advantage of server resources.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+4. When the job is complete, the results are represented in the front end.
+
+Register a Tripal Job
+-----------------------
+
+To register a job with the Tripal Jobs service we use the `tripal.job` service create method.
+
+As an example, consider the situation where your extension module provides a page to visualize a particular dataset. You have materialized views to allow you to provide the visualization within the page load, but you want to provide a link allowing users to download the dataset for further analysis.
+
+One way to do this is to create a service to prepare the dataset as a file but clearly that may not be feasible to do in the page load. Instead, you can create a button that runs the following snipped of code to register a job to complete this with Tripal. Then in your service, you create a public static method to setup and execute your service.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
One way to do this is to create a service to prepare the dataset as a file but clearly that may not be feasible to do in the page load. Instead, you can create a button that runs the following snippet of code to register a job to complete this with Tripal. Then in your service, you create a public static method to setup and execute your service.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+Callback
+^^^^^^^^^^
+
+Tripal 4 supports the following three types of callables for the callback parameter:
+
+ - **Simple function**: this is the method that was used in Tripal 2/3 where you provide a function that is usually in an include file or the module file. This approach is not recommend but still fully supported.
+ - **String-base Static Callback**: this method allows you to specify a static function in a string. You must include the full namespace, class and method. For example, `\Drupal\trpextension\Services\ExportVizData::runTripalJob`.
+ - **Array-based Static Callback**: this method is equivalent to the string-based approach but some feel is more readible. Here the callback is an array where the first element is the class and the second arguement is the name of the static method. For example, `[\Drupal\trpextension\Services\ExportVizData::class ,'runTripalJob']`.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
 - **Array-based Static Callback**: this method is equivalent to the string-based approach but some feel is more readible. Here the callback is an array where the first element is the class and the second arguement is the name of the static method. For example, `[\Drupal\trpextension\Services\ExportVizData::class, 'runTripalJob']`.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+You can `read more about PHP callables at php.net <https://www.php.net/manual/en/language.types.callable.php>`_.
+
+.. warning::
+
+  While php also supports passing an object paired with a non-static method name, Tripal Jobs does not support this as the callback is stored in the database. Additionally, an anonymous function is not recommend or tested for the same reasons.
+
+Get information about a job
+-----------------------------
+
+You may often find you also need the ability in your extension module to retrieve information about a specific job. Expanding on the previous example in registering a Tripal Job above, perhaps you want to check on the status of your users export job in order to let them know when it is complete and provide their results.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
You may often find you also need the ability in your extension module to retrieve information about a specific job. Expanding on the previous example in registering a Tripal Job above, perhaps you want to check on the status of your users' export job in order to let them know when it is complete and provide their results.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+When you created the tripal job before, you would have had the job_id returned to you. This unique identifier can be used to access all the information you need! You can store it in the form_state if you registered the job in multipage form or you can save it in the URL in the case of an export button redirecting to a results page.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
When you created the tripal job before, you would have had the job_id returned to you. This unique identifier can be used to access all the information you need! You can store it in the form_state if you registered the job in a multipage form or you can save it in the URL in the case of an export button redirecting to a results page.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  $status = $job->isRunning();
+  if ($status === TRUE) {
+    $progress == $job->getProgress();
+  }
+  // Or just check the status.
+  // This will return a human-readable string like 'Waiting', 'Cancelled',
+  // 'Running', 'Completed', or 'Error'.
+  $status = $job->getStatus();
+
+  // You can also retrieve the arguements that you passed into the job.
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
  // You can also retrieve the arguments that you passed into the job.
```

File: docs/dev_guide/module_dev/jobs.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  // We're grabbing the filename we asked the exported dataset to be saved in
+  // here when we find out the job is complete without error.
+  if ($status == 'Complete') {
+    $arguements = $job->getArguments();
+    $url = Url::fromUri($arguements['filename']);
+    $link = Link::fromTextAndUrl('access your results here', $url);
+    \Drupal::messenger()->addStatus("The export is complete and you can " . $link->toString());
```
Comment (dsenalik) [inline docs/dev_guide/module_dev/jobs.rst]:
```suggestion
    $arguments = $job->getArguments();
    $url = Url::fromUri($arguments['filename']);
    $link = Link::fromTextAndUrl('access your results here', $url);
    \Drupal::messenger()->addStatus("The export is complete and you can " . $link->toString());
```

**Review by dsenalik — APPROVED** [review]
Wonderful 😍 !
My usual picky suggestions but I will approve anyway

**Comment by laceysanderson** [comment]
Thanks for the spelling fixes and sorry there were more then there should have been 🙈

**Pull request merged**